src/apps/testapps/testPolyfill.c

```diff
@@ -59,6 +59,16 @@ Geofence transMeridianHoleGeofence;
 GeoPolygon transMeridianHoleGeoPolygon;
 GeoPolygon transMeridianFilledHoleGeoPolygon;
 
+static int countActualHexagons(H3Index* hexagons, int numHexagons) {
+    int actualNumHexagons = 0;
+    for (int i = 0; i < numHexagons; i++) {
+        if (hexagons[i] != 0) {
```

Comment:
I have been wondering where in the code do we rely on zero-initialization. There are reasons to avoid zero-initialization if possible. Is this a case when the caller relies on the fact that `hexagons` contains only valid hexagons or zeros?

Reply:
Yes. In algorithms where we return a list of hexagons and cannot say ahead of time how many we need, we require the caller to pass in a zero-filled array of the max possible size and then only fill H3 addresses for the real hexagons. We could theoretically fill in index order and then add a NULL as a stop signal afterward, but internally this isn't as efficient because we often need to use the array as a set, which we do via modulo arithmetic on the index, resulting in arbitrary index positions.